airflow/models/dagrun.py

```diff
@@ -124,7 +124,7 @@ class DagRun(Base, LoggingMixin):
         ),
     )
 
-    task_instances = relationship(TI, back_populates="dag_run")
+    task_instances = relationship(TI, back_populates="dag_run", cascade='all, delete, delete-orphan')
```

Comment:
According to documentation `all` implies `delete` so the latter is not needed. However, I don’t think we should do `expunge` (also implied by `all`). I’m also not sure about `refresh-expire`. Maybe a safer route would be `save-update, merge, delete, delete-orphan` (the first two are the default value).

https://docs.sqlalchemy.org/en/14/orm/relationship_api.html#sqlalchemy.orm.relationship.params.cascade